backends/mpi/world.hpp

```diff
@@ -54,11 +54,13 @@ class world : public bulk::world {
   int active_processors() const override final { return active_processors_; }
   int rank() const override final { return processor_id_; }
 
-  void sync(bool clear_queues = true) override final {
+  void sync(std::string tag = "", bool clear_queues = true) override final {
```

Comment:
This may break existing code that uses `clear_queues`, I see two options to avoid this:

1. Keep `clear_queues` as the first argument, making `tag` the second argument with a default.
2. Improve upon the existing `sync` by having a
   ```cpp
   struct sync_options {
      bool clear_queues = true;
      std::string tag = "";
   }
   ```
   which is arguably more future proof in case we want to add more options down the line. This would then be used like `world.sync({.tag = "foo"})`, which I like -- you can pass a subset of arguments as you please without us neading overloads for all of them, or explicitly passing values for aguments you don't care about.
  
   For backwards compatibility we would then still retain `sync(bool clear_queues)` which would forward to `sync({clear_queues: clear_queues})`

What do you think?

Reply:
@jwbuurlage yes you're right, the 2nd option would be better. I'll fix it ASAP